### PR TITLE
Add LocalDateDoubleTimeSeries.mapDates

### PR DESCRIPTION
**Pull request by cjkent**
Time series already have a `mapValues` method but there was no corresponding `mapDates`.

### CODE REVIEW

**Comment by cjkent** [comment]
retest this please

**Comment by cjkent** [comment]
Don't review or merge this yet, I've thought of something I need to check.

**Comment by cjkent** [comment]
The problem is the potential to reorder the dates. There are 3 ways we could deal with this

* Reorder the values - is this too subtle?
* Forbid reordering, throw an exception if the mapped dates aren't in ascending order
* Abandon the whole idea; this would be a shame as I have a concrete use case where it makes the calling code much nicer

**Comment by jodastephen** [comment]
It would have to be option 2, exception if reordered.

File: modules/collect/src/main/java/com/opengamma/strata/collect/timeseries/DenseLocalDateDoubleTimeSeries.java [hunk 1]
```diff
@@ -57,7 +62,7 @@
    * array for weekends and providing the different date
    * calculations for each case.
    */
-  static enum DenseTimeSeriesCalculation {
+  enum DenseTimeSeriesCalculation {
```
Comment (jodastephen) [inline modules/collect/src/main/java/com/opengamma/strata/collect/timeseries/DenseLocalDateDoubleTimeSeries.java]: It was clearer with the static (the code reader should not need to know offhand that nested enums are static by default)

Comment (cjkent) [inline modules/collect/src/main/java/com/opengamma/strata/collect/timeseries/DenseLocalDateDoubleTimeSeries.java]: They're not static by default, they're just static. Always.

Comment (jodastephen) [inline modules/collect/src/main/java/com/opengamma/strata/collect/timeseries/DenseLocalDateDoubleTimeSeries.java]: Agreed, but it is better for the source to be explicit, then it doesn't rely on esoteric knowledge of the code reader.

**Pull request merged**